crates/ethereum-rpc/src/gas_price/gas_oracle.rs

```diff
@@ -229,12 +230,14 @@ impl<C: sov_modules_api::Context> GasPriceOracle<C> {
             }
         }
         let last_entry = fee_entries.last().expect("is not empty");
+        let spec_id = fork_from_block_number(end_block_plus).spec_id;
```

Comment:
```suggestion
        let spec_id = fork_from_block_number(end_block_plus_one).spec_id;
```
`end_block_plus_one`, `next_block` or something